File: app/src/main/java/net/ginteam/carmen/contract/MapContract.java
package net.ginteam.carmen.contract;

import android.support.v7.app.AppCompatActivity;

import com.google.android.gms.maps.MapView;
import com.google.android.gms.maps.model.LatLng;
import com.google.android.gms.maps.model.LatLngBounds;

import net.ginteam.carmen.model.company.MapCompanyModel;

import java.util.List;

/**
 * Created by vadik on 10.01.17.
 */

public interface MapContract {

    interface View extends RequestLocationContract.View {

        void showGoogleMap(LatLng userLocation);

        MapView getMapView();

        AppCompatActivity getActivity();

        void animateToLocation(LatLng location);

        void showCompanies(List <MapCompanyModel> companies);

        void showSearchControl(boolean show);

        void showCompaniesControls(boolean show);

    }

    interface Presenter extends RequestLocationContract.Presenter<View> {

        void getLastUserLocation();

        void fetchCompaniesByBounds(int categoryId, String filters, LatLngBounds bounds);

    }

}

File: app/src/main/kotlin/net/ginteam/carmen/kotlin/contract/MapContract.kt
package net.ginteam.carmen.kotlin.contract

import com.google.android.gms.maps.model.LatLng
import com.google.android.gms.maps.model.LatLngBounds
import net.ginteam.carmen.kotlin.model.CategoryModel

/**
 * Created by eugene_shcherbinock on 2/21/17.
 */

object MapActivityContract {

    interface View : LocationContract.View {

        fun showCompaniesMapFragment(startLocation: LatLng)

    }

    interface Presenter : LocationContract.Presenter <View>

}

object MapCompaniesFragmentContract {

    interface View : BaseCompaniesContract.View {

        fun showSearchView(show: Boolean)
        fun showCompaniesView(show: Boolean)

    }

    interface Presenter : BaseCompaniesContract.Presenter <View> {

        fun fetchCompanies(category: CategoryModel, filterQuery: String, mapBounds: LatLngBounds)

    }

}

object MapCompanyFragmentContract {

    interface View : BaseContract.View

    interface Presenter : BaseContract.Presenter <View>

}